wave_vision/src/dataset.cpp

```diff
@@ -0,0 +1,255 @@
+#include "wave/vision/dataset.hpp"
+
+namespace wave {
+
+bool TestCamera::update(double dt) {
+    this->dt += dt;
+
+    if (this->dt > (1.0 / this->hz)) {
+        this->dt = 0.0;
+        this->frame++;
+        return true;
+    }
+
+    return false;
+}
+
+int TestCamera::checkFeatures(double dt,
+                              const MatX &features,
+                              const Vec3 &rpy,
+                              const Vec3 &t,
+                              std::vector<std::pair<Vec2, Vec3>> &observed) {
+    Vec3 f_2d, rpy_edn, t_edn;
+    std::pair<Vec2, Vec3> obs;
+    Vec4 f_3d, f_3d_edn;
+    Mat3 R;
+    MatX P;
+
+    // pre-check
+    if (this->update(dt) == false) {
+        return 1;
+    }
+
+    // rotation matrix - convert from nwu to edn then to rotation matrix R
```

Comment:
ah okay. We've talked before about NWU being the standard. I was more curious about why the change to EDN - Gazebo makes sense.

Hmm, up to you. I don't think it makes too much of a difference. As long as we (and the future users) know to keep track of everything it should be okay. Coordinate systems can quickly mess up a lot of things!